### DISCUSSION TITLE
0.23.0 JSVM onBootstrap without e.next() causes segfault

### DISCUSSION THREAD

**Discussion by dschissler** [pocketbase/pocketbase#5612] [Q&A] [answered]
This causes a segfault:

```js
onBootstrap((e) => {
  console.log("App initialized!")
})
```

This doesn't:

```js
onBootstrap((e) => {
  console.log("App initialized!")

  e.next()
})
```

This took me a while to figure out.  It was confusing because I wouldn't expect a segfault there, and so I was looking at my other custom Golang stuff to get to that point.  I probably would have figured it out a lot more quickly if it had non-fatally hung after displaying that.

**Comment by ganigeorgiev** [comment, marked as answer]
This is expected because `e.next()` is always required to be invoked by the hooks, otherwise the hook can't proceed, aka. if users don't call `e.next()` this implies that they want to override the default hook chain execution with a custom implementation.

In your case you are essentially stopping the default app bootstrap process without doing anything and it is probably failing with `nil` access error or other similar (it is the same if you try to access the db for example with the old version in `onBeforeBootstrap`).

_Specifically for the bootstrap hook I guess we can add an extra call after the hook trigger and check manually whether `$app.isBootstrapped()` is true and if not - print a more user friendly message._

Note that if you want to execute something AFTER the app has been bootstrapped you need to call `e.next()` first:

```js
onBootstrap((e) => {
  e.next()

  // ...
})
```

**Reply by dschissler** [reply]
> Specifically for the bootstrap hook I guess we can add an extra call after the hook trigger and check manually whether $app.isBootstrapped() is true and if not - print a more user friendly message.

This could be very helpful to remind people to do it for all of  their events.  You could add an education message there since it's probably one of the first things to mess up while upgrading, and for new users.

In my case there is a `main.pb.js` file with only that init message and nothing else.  So it was very easy to overlook it.
